examples/apps/cli/main.c

```diff
@@ -108,12 +152,43 @@ int main(int argc, char *argv[])
     otDiagInit(sInstance);
 #endif
 
+    /* Register thread state change handler */
+    otSetStateChangedCallback(sInstance, handleNetifStateChanged, sInstance);
+
+    /* Register GPIO button handlers for first 2 buttons */
+    otSysGpioRegisterCallback(BUTTON_GPIO_PORT, BUTTON_1_PIN, handleGpioInterrupt1, sInstance);
+    otSysGpioIntEnable(BUTTON_GPIO_PORT, BUTTON_1_PIN);
+
+    otSysGpioRegisterCallback(BUTTON_GPIO_PORT, BUTTON_2_PIN, handleGpioInterrupt2, sInstance);
+    otSysGpioIntEnable(BUTTON_GPIO_PORT, BUTTON_2_PIN);
+
+    /* Disable ICMP Echo handling so we can process it from the ip6 handler.
```

Comment:
I think this example will be a lot simpler/cleaner if we just keep UDP and remove ICMPv6 Echo Request. UDP is also the preferred path for applications to use. Thoughts?

Reply:
If that is preferred, I have no problem just sticking with UDP.  That will shorten the Codelab a little bit too which is always a good thing.

Reply:
Having looked at the code, I think the UDP-only path is overall simpler and cleaner. I realize now there's a number of hoops necessary to support ICMPv6. Also, users on the forums generally want to see examples of UDP and/or CoAP, not really how to intercept ICMPv6 packets.

Reply:
SGTM, let's do that then